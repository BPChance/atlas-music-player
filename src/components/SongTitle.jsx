export default function SongTitle() {
  return (
    <div className="flex flex-col">
      <h1 className="text-2xl font-bold">Tidal Drift</h1>
      <p className="text-base">Echoes of the Sea</p>
    </div>
  );
}
