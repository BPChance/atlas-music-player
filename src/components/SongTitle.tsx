type SongTitleProps = {
  title: string;
  artist: string;
};

export default function SongTitle({ title, artist }: SongTitleProps) {
  return (
    <div className="w-full text-left">
      <h1 className="text-supapink text-2xl font-bold">{title}</h1>
      <p className="text-base text-[#94A3B8]">{artist}</p>
    </div>
  );
}
