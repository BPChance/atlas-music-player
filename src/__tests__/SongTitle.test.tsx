import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import SongTitle from "../components/SongTitle";

describe("SongTitle", () => {
  it("renders a song title and artist", () => {
    const { container } = render(
      <SongTitle title="test title" artist="test artist" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with empty artist", () => {
    const { container } = render(<SongTitle title="what artist?" artist="" />);
    expect(container).toMatchSnapshot();
  });
});
