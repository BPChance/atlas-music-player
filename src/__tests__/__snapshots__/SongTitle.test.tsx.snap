// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders a song title and artist 1`] = `
<div>
  <div
    class="w-full text-left"
  >
    <h1
      class="text-supapink text-2xl font-bold"
    >
      test title
    </h1>
    <p
      class="text-base text-[#94A3B8]"
    >
      test artist
    </p>
  </div>
</div>
`;

exports[`SongTitle > renders correctly with empty artist 1`] = `
<div>
  <div
    class="w-full text-left"
  >
    <h1
      class="text-supapink text-2xl font-bold"
    >
      what artist?
    </h1>
    <p
      class="text-base text-[#94A3B8]"
    />
  </div>
</div>
`;
